sessionctx/stmtctx/stmtctx.go

```diff
@@ -199,6 +199,8 @@ type StatementContext struct {
 	// InVerboseExplain indicates the statement is "explain format='verbose' ...".
 	InVerboseExplain bool
 
+	// EnableOptimizeTrace indicates whether enable optimizer trace by 'trace plan statement'
+	EnableOptimizeTrace bool
```

Comment:
only optimizing plan avoided `ResetContextOfStmt`, thus we can pass the flag by stmtCtx now.